mergify_engine/actions/queue.py

```diff
@@ -146,6 +149,11 @@ async def _should_be_merged(self, ctxt: context.Context, q: queue.Queue) -> bool
 
     async def _should_be_synced(self, ctxt: context.Context, q: queue.Queue) -> bool:
         # NOTE(sileht): since we create dedicated branch we don't need to sync PR
+        if await ctxt.is_behind and await q.is_first_pull(ctxt):
```

Comment:
```suggestion
        if await q.is_first_pull(ctxt) and await ctxt.is_behind:
```

neat, but I guess `is_first_pull` is cheaper to execute :)

Reply:
Approving if you think it's not worth it, remove the label